vcd/resource_vcd_lb_server_pool.go

```diff
@@ -0,0 +1,346 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdLBServerPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLBServerPoolCreate,
+		Read:   resourceVcdLBServerPoolRead,
+		Update: resourceVcdLBServerPoolUpdate,
+		Delete: resourceVcdLBServerPoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLBServerPoolImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the Service Monitor is located",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the Service Monitor is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the Server Pool is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Server Pool name",
+			},
+
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server pool description",
+			},
+			"algorithm": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Balancing method for the service. One of ip-hash, round-robin, uri, leastconn, url, or httpheader",
+				ValidateFunc: validateCase("lower"),
```

Comment:
Is this the only validation required? No check on a list of pre-defined values?

Reply:
I rely on the API error which looks the following for the user:

```
vcd_lb_server_pool.web-servers: Modifying... [id=pool-94]

Error: unable to update load balancer server pool with ID pool-94: error while updating load balancer server pool : vShield Edge [LoadBalancer] Invalid algorithm asd. Valid algorithms are: IP-HASH|ROUND-ROBIN|URI|LEASTCONN|URL|HTTP-HEADER. (API error: 14553)
```

There are ugly things about it though:
* One - the error suggests capital cased names while it returns lower cased ones
* Two - it suggests `HTTP-HEADER`, but in fact it is `httpheader`

On the other hand. If I hardcode it in Terraform validator - if future versions introduce new field - we will have to alter it in code.